tests/neg/i16453.check

```diff
@@ -0,0 +1,8 @@
+-- [E172] Type Error: tests/neg/i16453.scala:10:43 ---------------------------------------------------------------------
+10 |  val fails = summon[String => Option[Int]] // error
+   |                                           ^
+   |No given instance of type String => Option[Int] was found for parameter x of method summon in object Predef
+   |
+   |Note: Chaining implicit conversions is no longer allowed in Scala. The following conversions were ignored:
```

Comment:
As far as I know, scala has never chained implicits implicitly - if this was intended , one had to add an implicit parameter to the definition of the other implicit, so `no longer` seems wrong to me here (the description https://github.com/lampepfl/dotty/issues/16453 is quite misleading). And from the description of this PR I understand that the issue is not only about chaining conversions but in general applying an implicit conversion to a value being the result of an implicit search - the fact that the implicit value to be found in this example is a function seems to overcomplicate things here. As far as I understand your changes should also get applied in a situation like this
```scala
import scala.language.implicitConversions

object Test {
  def test() = {
    implicit def int: Int = 0
    implicit def toOption[T](t: T): Option[T] = Option(t)

    val x = implicitly[Option[Int]](/* implicitly[Int] */)
  }
}
```

Reply:
Oh, I didn't realize Scala 2 didn't chain implicits either. I pretty much copied the example error message in #16453 without thinking too much about it, I can simplify the test and add a check for implicitly as you suggested later.